busted/outputHandlers/json.lua

```diff
@@ -14,12 +14,26 @@ return function(options)
       errors = handler.errors,
       duration = handler.getDuration()
     }
-    local ok, result = pcall(json.encode, error_info)
+
+    for _, test in ipairs(handler.pendings) do
+      test.element.attributes.default_fn = nil -- functions cannot be encoded into json
+    end
+
+    local ok, result = pcall(json.encode, error_info, {
+      exception = function(reason, value, state, default_reason)
+        local state_short = table.concat(state.buffer, '')
+        state_short = ('... %s %s'):format(state_short:sub(#state_short - 100), tostring(state.exception))
+
+        local err_msg = (default_reason .. ' in (%s)\n'):format(state_short)
+        io.stderr:write(err_msg)
+      end,
+    })
 
     if ok then
       io_write(result)
     else
       io_write("Failed to encode test results to json: " .. result)
+      os.exit(1)
```

Comment:
Why write an error message and then exit? Shouldn't this be an error? Either `error()` or `io.stderr:write(...); os.exit(1)`. Or is there a reason this written like this?

Reply:
You are right, it is redundant.  I added it before the exception callback of json.encode, where stderr is set.
io_write here is to also write the error message to stdout, for cases when json output is redirected to a file.

Reply:
My first instinct was `error()` too, but I did see there was some instances of the other construct already. I saw them on a grep but didn't look into *why* there might be that way. It's probably worth checking and comparing use cases.